### ISSUE TITLE
Scan element detection timeout set to 600 seconds. How to change it?

### ISSUE THREAD

**Issue by zedtux** [fastlane/fastlane#12711]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When I run `bundle exec fastlane ios tests`, which is using scan :

```ruby
platform :ios do
  desc 'Runs all the tests'
  lane :tests do
    scan(
      scheme: 'Companion',
      project: './ios/Companion.xcodeproj'
    )
  end
end
```

It is taking a lot of time to end up with the following message :

```
All tests
Test Suite CompanionTests.xctest started
CompanionTests
✗ testRendersWelcomeScreen, ((foundElement) is true) failed - Couldn't find element with text 'Welcome to React Native!' in 600 seconds


CompanionTests
testRendersWelcomeScreen, ((foundElement) is true) failed - Couldn't find element with text 'Welcome to React Native!' in 600 seconds
/Users/guillaumehain/Developments/companion/ios/CompanionTests/CompanionTests.m:64
```
  XCTAssertNil(redboxError, @"RedBox error: %@", redboxError);
  XCTAssertTrue(foundElement, @"Couldn't find element with text '%@' in %d seconds", TEXT_TO_LOOK_FOR, TIMEOUT_SECONDS);
}
  ```


	 Executed 1 test, with 1 failure (0 unexpected) in 600.062 (600.063) seconds
```

600 seconds is really too much. I looked at the scan documentation but didn't found any way to change this value to something like 30 seconds.

Is there anything already existing or does it have to be implemented?

##### Complete output when running fastlane, including the stack trace and command used
<details>
  <pre>All tests
Test Suite CompanionTests.xctest started
CompanionTests
✗ testRendersWelcomeScreen, ((foundElement) is true) failed - Couldn't find element with text 'Welcome to React Native!' in 600 seconds

CompanionTests
testRendersWelcomeScreen, ((foundElement) is true) failed - Couldn't find element with text 'Welcome to React Native!' in 600 seconds
/Users/guillaumehain/Developments/companion/ios/CompanionTests/CompanionTests.m:64
```
  XCTAssertNil(redboxError, @"RedBox error: %@", redboxError);
  XCTAssertTrue(foundElement, @"Couldn't find element with text '%@' in %d seconds", TEXT_TO_LOOK_FOR, TIMEOUT_SECONDS);
}
  ```

	 Executed 1 test, with 1 failure (0 unexpected) in 600.062 (600.063) seconds

** TEST FAILED **
[15:01:34]: Exit status: 65
[15:01:34]: Successfully sent Slack notification
+--------------------+---+
|      Test Results      |
+--------------------+---+
| Number of tests    | 1 |
| Number of failures | 1 |
+--------------------+---+</pre>
</details>

### Environment

<details>
  <pre><details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                         |
| --------------------------- | --------------------------------------------- |
| OS                          | 10.13.6                                       |
| Ruby                        | 2.4.1                                         |
| Bundler?                    | true                                          |
| Git                         | git version 2.15.1 (Apple Git-101)            |
| Installation Source         | ~/.rvm/gems/ruby-2.4.1@companion/bin/fastlane |
| Host                        | Mac OS X 10.13.6 (17G31f)                     |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.4.1/lib                  |
| OpenSSL Version             | OpenSSL 1.1.0h  27 Mar 2018                   |
| Is contained                | false                                         |
| Is homebrew                 | false                                         |
| Is installed via Fabric.app | false                                         |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/   |
| Xcode Version               | 9.4                                           |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     |             |   |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version '2.96.1'

default_platform(:android)

before_all do
  update_fastlane
  yarn(
    command: 'install',
    package_path: './package.json'
  )
  ANDROID_VERSION_NAME = get_version_name(
    gradle_file_path: 'android/app/build.gradle',
    ext_constant_name: 'versionName'
  )
  ANDROID_VERSION_CODE = get_version_code(
    gradle_file_path: 'android/app/build.gradle',
    ext_constant_name: 'versionCode'
  )
end

platform :ios do
  desc 'Runs all the tests'
  lane :tests do
    scan(
      scheme: 'Companion',
      project: './ios/Companion.xcodeproj'
    )
  end
end

platform :android do
  desc 'Runs all the tests'
  lane :tests do
    gradle(task: 'assemble', project_dir: './android/')

    sh '$(which adb) devices'
    sh 'pwd'
    sh 'ls -al ../android/app/build/outputs/apk/'

    gradle(task: 'test', project_dir: './android/')
    sh 'cd .. && $(which bundle) exec calabash-android run android/app/build/outputs/apk/app-debug.apk'
  end

  private_lane :clean_android do
    gradle(task: 'clean', project_dir: './android/')
  end

  desc 'Submit a new Beta Build to Crashlytics Beta'
  lane :beta do
    gradle(task: 'clean assembleRelease')
    crashlytics

    # sh "your_script.sh"
    # You can also use other beta testing services here
  end

  desc 'Deploy a new version to the Google Play'
  lane :deploy do |options|
    clean_android

    gradle(task: 'assemble', build_type: 'release', project_dir: './android/')

    if options[:play_store]
      # upload to alpha track on google play developer's console
      supply(
        track: 'alpha',
        apk: 'android/app/build/outputs/apk/app-release.apk'
      )
    end
  end
end

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.97.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                           | Version | Update-Status |
| -------------------------------- | ------- | ------------- |
| fastlane-plugin-badge            | 1.1.0   | ✅ Up-To-Date  |
| fastlane-plugin-get_version_code | 0.2.0   | ✅ Up-To-Date  |
| fastlane-plugin-get_version_name | 0.2.2   | ✅ Up-To-Date  |
| fastlane-plugin-yarn             | 1.0     | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                              | Version      |
| -------------------------------- | ------------ |
| did_you_mean                     | 1.1.0        |
| io-console                       | 0.4.6        |
| CFPropertyList                   | 2.3.6        |
| public_suffix                    | 2.0.5        |
| addressable                      | 2.5.2        |
| atomos                           | 0.1.2        |
| awesome_print                    | 1.8.0        |
| babosa                           | 1.0.2        |
| curb                             | 0.9.6        |
| fastimage                        | 2.1.3        |
| bundler                          | 1.16.2       |
| colored                          | 1.2          |
| highline                         | 1.7.10       |
| commander-fastlane               | 4.4.6        |
| dotenv                           | 2.4.0        |
| emoji_regex                      | 0.1.1        |
| excon                            | 0.62.0       |
| multipart-post                   | 2.0.0        |
| faraday                          | 0.15.2       |
| unf_ext                          | 0.0.7.5      |
| unf                              | 0.1.4        |
| domain_name                      | 0.5.20180417 |
| http-cookie                      | 1.0.3        |
| faraday-cookie_jar               | 0.0.6        |
| faraday_middleware               | 0.12.2       |
| gh_inspector                     | 1.1.3        |
| jwt                              | 2.1.0        |
| little-plugger                   | 1.1.4        |
| multi_json                       | 1.13.1       |
| logging                          | 2.2.2        |
| memoist                          | 0.16.0       |
| os                               | 0.9.6        |
| signet                           | 0.8.1        |
| googleauth                       | 0.6.2        |
| httpclient                       | 2.8.3        |
| mime-types-data                  | 3.2016.0521  |
| mime-types                       | 3.1          |
| declarative                      | 0.0.10       |
| declarative-option               | 0.1.0        |
| uber                             | 0.1.0        |
| representable                    | 3.0.4        |
| retriable                        | 2.0.2        |
| google-api-client                | 0.21.2       |
| json                             | 1.8.6        |
| mini_magick                      | 4.5.1        |
| multi_xml                        | 0.6.0        |
| plist                            | 3.4.0        |
| rubyzip                          | 1.2.1        |
| security                         | 0.1.3        |
| naturally                        | 2.2.0        |
| simctl                           | 1.6.4        |
| slack-notifier                   | 2.3.2        |
| terminal-notifier                | 1.8.0        |
| unicode-display_width            | 1.4.0        |
| terminal-table                   | 1.8.0        |
| tty-screen                       | 0.6.4        |
| tty-cursor                       | 0.5.0        |
| tty-spinner                      | 0.8.0        |
| word_wrap                        | 1.0.0        |
| claide                           | 1.0.2        |
| colored2                         | 3.1.2        |
| nanaimo                          | 0.2.5        |
| xcodeproj                        | 1.5.9        |
| rouge                            | 2.0.7        |
| xcpretty                         | 0.2.8        |
| xcpretty-travis-formatter        | 1.0.0        |
| badge                            | 0.9.1        |
| builder                          | 3.2.3        |
| gherkin                          | 4.1.3        |
| cucumber-core                    | 1.5.0        |
| cucumber-wire                    | 0.0.1        |
| diff-lcs                         | 1.3          |
| multi_test                       | 0.1.2        |
| cucumber                         | 2.99.0       |
| escape                           | 0.0.4        |
| thor                             | 0.20.0       |
| luffa                            | 2.0.0        |
| slowhandcuke                     | 0.0.3        |
| calabash-android                 | 0.9.2        |
| clipboard                        | 1.1.2        |
| edn                              | 1.1.1        |
| geocoder                         | 1.4.9        |
| command_runner_ng                | 0.1.3        |
| concurrent-ruby                  | 1.0.5        |
| i18n                             | 0.9.5        |
| run_loop                         | 2.7.1        |
| calabash-cucumber                | 0.21.5       |
| fastlane-plugin-badge            | 1.1.0        |
| fastlane-plugin-get_version_code | 0.2.0        |
| fastlane-plugin-get_version_name | 0.2.2        |
| fastlane-plugin-yarn             | 1.0          |
</details>

*generated on:* **2018-06-07**
</details></pre>
</details>

**Comment by zedtux** [comment]
Come one stupid bot, there isn't activity from Fastlane team, I'm waiting on this issue !

You should promote my issue to the team instead of try to close it : Got to hell.

**Comment by zedtux** [comment]
🤦‍♂️

**Comment by zedtux** [comment]
@fastlane-bot uninstall yourself.

**Comment by zedtux** [comment]
Yeah ... `waiting-for-reply` from Fastlane author ... what shall I do in order to get this issue closed by this stupid robot?

**Comment by zedtux** [comment]
Yeah, congrats 👍, good job bot, you're smart, our future is really sexy.

And thank you very much to the fastlane guys for the grateful help, my problem is now fixed thanks to your bot 💪

.............................. 🤦‍♂️

**Comment by GoMino** [comment]
@zedtux I guess your are using react-native, if so you need to change a const define in your test file 
`#define TIMEOUT_SECONDS 600`

Usually the react native project is ootstrap with a test target and this constant is defined in a file name `YourProject/YourProjectTest.m`
